nixos/modules/services/hardware/intel-aesmd.nix

```diff
@@ -0,0 +1,43 @@
+{ config, lib, pkgs, utils, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.intel-aesmd;
+
+in {
+
+  options.services.intel-aesmd = {
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Enable Intel's Architectural Enclave Service Manager for Intel SGX
+      '';
+    };
+  };
+
+  options.services.intel-aesmd.package = mkOption {
+    type = types.package;
+    default = pkgs.intel-aesmd;
+    defaultText = "pkgs.intel-aesmd";
+    description = ''
+      aesmd derivation to use.
+    '';
+  };
+
+  config = mkIf cfg.enable {
+    users.groups.aesmd = {
+    };
+
+    users.users.aesmd = {
+      description = "aesmd user";
+    };
+
+    environment.systemPackages = [ cfg.package ];
+    systemd.packages = [ cfg.package ];
+    systemd.services.aesmd.wantedBy = [ "multi-user.target" ];
```

Comment:
I kind of doubt this would work (or it might need some tweaking) but it would be awesome if `DynamicUser` worked for this

Reply:
First time I read about `DynamicUser`. It seems to me this could be done. There's a unix socket that needs to be exposed in `/run` with `RuntimeDirectory`, and the daemon needs access to `/dev/isgx`. I can look into this, but perhaps in an upcoming PR?